### ISSUE TITLE
MA0165: Make interpolated string

### ISSUE THREAD

**Issue by DmitriyMaksimov** [meziantou/Meziantou.Analyzer#764]
### Version of the Meziantou.Analyzer NuGet package

2.0.175

### Rule Identifier

MA0165

### Target Framework

net8.0

### C# Language version

C# 10

### Description

After upgrading to `2.0.175` from `2.0.173` getting the following errors
```
error MA0165: Make interpolated string (https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0165.md) [/home/ec2-user/workspace/xxx.csproj::TargetFramework=net8.0]
```

Required change doesn't make much sense - the analyzer requires converting all strings to interpolated strings

<img width="447" alt="image" src="https://github.com/user-attachments/assets/20b9d707-245c-440d-bdb5-ed0f60dd68fc">

<img width="481" alt="image" src="https://github.com/user-attachments/assets/b9d2fac3-b230-4049-9a05-149b000bb99b">

Also, the link `https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0165.md` is broken

### Reproduction Steps

Minimal code:

```c#
	public class Query
	{
		[JsonPropertyName("string")]
		public string Str { get; set; }
        }
```

**Comment by meziantou** [comment]
The default severity is `Hidden` (silent), which means that your IDE shouldn't report it in the error window or in the text editor.

This rule is only intended to help developers.

Nothing visible:

![image](https://github.com/user-attachments/assets/c943fc14-4cce-4607-839e-b7f9c4ccf57b)

You get a refactoring (similar to the existing one to convert to raw string)

<img width="751" src="https://github.com/user-attachments/assets/bc01498d-b7e7-43d1-96ad-e6a79b4e21ee">

For the documentation, I'll add it (and check why the CI didn't catch it)

**Comment by MuffPotter** [comment]
Same issue with Visual Studio 17.11.5:
[assembly: AssemblyTitle("Draeger.Bridge.Framework.XLiff")]

Only using dotnet_diagnostic.MA0165.severity=suggestion fixed the compiler error

**Comment by meziantou** [comment]
`dotnet_diagnostic.MA0165.severity=suggestion` would make it visible in the IDE, which is not intended. I do not reproduce the issue on my side (using VS 17.12 preview 4).

- Could you share a small repro?
- Which version of Roslyn are you using? (https://www.meziantou.net/getting-roslyn-csharp-compiler-and-language-versions.htm)
- Is it better if you force it to silent? `dotnet_diagnostic.MA0165.severity=silent`

> fixed the compiler error

Which compiler error? Do you have an error when building the project?

**Comment by DmitriyMaksimov** [comment]
Thank you very much for a such quick reply, @meziantou! 🙏 
And thank you for absolutely great project! Your analyzer is what I add to all C# projects - IMO it's "must have" 👍 

You are right - I have the following in `.editorconfig` file
```
# default severity for all analyzer diagnostics
dotnet_analyzer_diagnostic.severity = error
```

It's why I'm getting errors. It's a bit surprising that the default has a higher priority than defined in the analyzer.

Adding the following `.editorconfig`
```
dotnet_diagnostic.MA0165.severity = none
```

resolves the issue.

BTW, Rider has the same code fix for a long time ("To string interpolation"), so for Rider users it's redundant (and confusing which one to use 😁). But for VS users, it's a great code fix (for such users, the severity should be set to `silent`)! 👍 

<img width="350" alt="image" src="https://github.com/user-attachments/assets/a61330bd-3e68-4d47-9879-57fdc1cd1458">

**Comment by DmitriyMaksimov** [comment]
FYI: Rider is now [free](https://blog.jetbrains.com/blog/2024/10/24/webstorm-and-rider-are-now-free-for-non-commercial-use/) for non-commercial use - so you can use it for your project.

**Comment by fforjan** [comment]
Same for us, our default severity is error, came here so we had to do the
`dotnet_diagnostic.MA0165.severity = none`
